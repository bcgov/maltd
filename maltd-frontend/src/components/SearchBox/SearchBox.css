.search-box {
  width: 200px;
}
