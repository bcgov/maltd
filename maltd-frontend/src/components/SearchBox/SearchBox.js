/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { Button } from "reactstrap";
import PropTypes from "prop-types";
import "./SearchBox.css";

export default function SearchBox({
  searchBox: { type, color, disabled, block, active, outline }
}) {
  return (
    <Button
      className="search-box"
      type={type}
      color={color}
      disabled={disabled}
      block={block}
      active={active}
      outline={outline}
    >
      Search
    </Button>
  );
}

SearchBox.propTypes = {
  searchBox: PropTypes.shape({
    type: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired,
    block: PropTypes.bool.isRequired,
    active: PropTypes.bool.isRequired,
    outline: PropTypes.bool.isRequired
  }).isRequired
};
