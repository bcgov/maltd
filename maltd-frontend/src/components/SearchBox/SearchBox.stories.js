/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import SearchBox from "./SearchBox";

const searchBox = {
  type: "submit",
  color: "success",
  disabled: true,
  block: false,
  active: false,
  outline: false
};

storiesOf("SearchBox", module)
  .add("default", () => <SearchBox searchBox={searchBox} />)
  .add("valid", () => (
    <SearchBox searchBox={{ ...searchBox, disabled: false }} />
  ))
  .add("invalid", () => (
    <SearchBox searchBox={{ ...searchBox, color: "danger" }} />
  ));
