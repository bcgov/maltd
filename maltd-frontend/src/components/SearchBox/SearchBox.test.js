/* eslint-disable react/jsx-filename-extension */
import React from "react";
import renderer from "react-test-renderer";
import SearchBox from "./SearchBox";

describe("Search Box", () => {
  test("Search box component renders as expected", () => {
    const component = renderer.create(
      <SearchBox
        searchBox={{
          type: "submit",
          color: "success",
          disabled: true,
          block: false,
          active: false,
          outline: false
        }}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
