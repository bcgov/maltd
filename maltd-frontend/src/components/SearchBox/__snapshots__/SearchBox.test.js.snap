// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field Input field component renders as expected 1`] = `
<input
  className="input-field form-control"
  name="myinputfield"
  placeholder="placeholder"
  type="text"
  value="idir"
/>
`;
